src/titan_checkpoint_impl.cc

```diff
@@ -194,195 +195,78 @@ Status TitanCheckpointImpl::CreateCustomCheckpoint(
         uint64_t* sequence_number, uint64_t log_size_for_flush,
         const std::string full_private_path) {
   Status s;
-  std::vector<std::string> live_files;
-  std::vector<std::string> titan_live_files;
+  std::vector<std::string> titandb_files;
   std::vector<VersionEdit> version_edits;
-  uint64_t base_manifest_file_size = 0;
-  uint64_t min_log_num = port::kMaxUint64;
-  *sequence_number = db_->GetLatestSequenceNumber();
   bool same_fs = true;
-  VectorLogPtr live_wal_files;
 
-  bool flush_memtable = true;
-  if (!db_options.allow_2pc) {
-    if (log_size_for_flush == port::kMaxUint64) {
-      flush_memtable = false;
-    } else if (log_size_for_flush > 0) {
-      // if out standing log files are small, we skip the flush.
-      s = db_->GetSortedWalFiles(live_wal_files);
-
-      if (!s.ok()) {
-        return s;
-      }
-
-      // Don't flush column families if total log size is smaller than
-      // log_size_for_flush. We copy the log files instead.
-      // We may be able to cover 2PC case too.
-      uint64_t total_wal_size = 0;
-      for (auto& wal : live_wal_files) {
-        total_wal_size += wal->SizeFileBytes();
-      }
-      if (total_wal_size < log_size_for_flush) {
-        flush_memtable = false;
-      }
-      live_wal_files.clear();
-    }
-  }
-
-  // This will return live files prefixed with "/"
-  s = db_->GetTitanLiveFiles(live_files, &base_manifest_file_size,
-                              &version_edits, flush_memtable);
+  // Create base db checkpoint
+  auto base_db_checkpoint = new rocksdb::CheckpointImpl(db_);
+  s = base_db_checkpoint->CreateCustomCheckpoint(db_options, link_file_cb, 
```

Comment:
it should be called between `DisableFileDeletions` and `EnableFileDeletions`

Reply:
On the outside of `TitanCheckpointImpl::CreateCustomCheckpoint`, we've already called `DisableFileDeletions ` and `EnableFileDeletions `. I think we can skip it here.

Reply:
Then is it necessary to call `DisableFileDeletions` and `EnableFileDeletions` again in `GetAllTitanFiles`?

Reply:
`GetAllTitanFiles` may be called by other functions in the future, but I think `CreateCustomCheckPoint` will only be used by `CreateCheckPoint`.

Reply:
Would someone will call `CreateCustomCheckpoint` directly? If that, the behavior would be broken.

Reply:
Done.

Reply:
Seems the `EnableFileDeletions` shoud be called after `GetAllTitanFiles`?

Reply:
And we don't need to call `DisableFileDeletions` and `EnableFileDeletions` in `CreateCheckpoint` anymore.

Reply:
Oh, that's true, I fixed it.